Makefile

```diff
@@ -50,109 +51,127 @@ export CGO_ENABLED=0
+	echo "-----------------\\n" &&\\
+	git status &&\\
+	echo "\\n-----------------" &&\\
+	echo "NOTICE: There are some files that have not been committed." &&\\
+	echo "-----------------\\n"  &&\\
+	exit 0)
 
+## verify: Run format and lint checks
 verify: verify-format lint
 
-generate: install-mockgen
-	@echo "> Generating mocks..."
-	$(GOCMD) generate ./...
-
+## verify-format: Verify the format
 verify-format: install-goimports
-	@echo "> Verifying format..."
+	@echo "=====> Verifying format..."
 	$(if $(shell goimports -l -local ${PACKAGE_BASE} ${SRC}), @echo ERROR: Format verification failed! && goimports ${GOIMPORTS_DIFF_OPTION} -local ${PACKAGE_BASE} ${SRC} && exit 1)
 
-prepare-for-pr: tidy verify test
-	@git diff-index --quiet HEAD -- ||\
-	(echo "-----------------" &&\
-	echo "NOTICE: There are some files that have not been committed." &&\
-	echo "-----------------\n" &&\
-	git status &&\
-	echo "\n-----------------" &&\
-	echo "NOTICE: There are some files that have not been committed." &&\
-	echo "-----------------\n"  &&\
-	exit 0)
-
+## benchmark: Run benchmark tests
 benchmark: out
-	@echo "> Running Benchmarks"
+	@echo "=====> Running benchmarks"
 	$(GOCMD) test -run=^$  -bench=. -benchtime=1s -benchmem -memprofile=./out/bench_mem.out -cpuprofile=./out/bench_cpu.out -tags=benchmarks ./benchmarks/ -v
-# NOTE: You can analyze the results, using go tool pprof. For instance, you can start a server to see a graph of the cpu usage by running
-# go tool pprof -http=":8082" out/bench_cpu.out. Alternatively, you can run go tool pprof, and in the ensuing cli, run
-# commands like top10 or web to dig down into the cpu and memory usage
-# For more, see https://blog.golang.org/pprof
 
-# NOTE: Windows doesn't support `-p`
+## package: Package the program
+package: out
+	tar czf .$/out$/$(ARCHIVE_NAME).tgz -C .$/out$/ $(PACK_BIN)
+
+## install: Install the program to the system
+install:
+	@mkdir -p ${DESTDIR}${BINDIR}
+	@cp ./out/$(PACK_BIN) ${DESTDIR}${BINDIR}/
+
+## install-mockgen: Used only by apt-get install when installing ubuntu ppa
+install-mockgen:
+	@echo "=====> Installing mockgen..."
+	cd tools && $(GOCMD) install github.com/golang/mock/mockgen
+
+## install-goimports: Install goimports dependency
+install-goimports:
+	@echo "=====> Installing goimports..."
+	cd tools && $(GOCMD) install golang.org/x/tools/cmd/goimports
+
+## install-golangci-lint: Install golangci-lint dependency
+install-golangci-lint:
+	@echo "=====> Installing golangci-lint..."
+	cd tools && $(GOCMD) install github.com/golangci/golangci-lint/cmd/golangci-lint
+
+## mod-tidy: Tidy Go modules
+mod-tidy:
+	$(GOCMD) mod tidy  -compat=1.19
+	cd tools && $(GOCMD) mod tidy -compat=1.19
+
+## tidy: Tidy modules and format the code
+tidy: mod-tidy format
+
+## out: Make a directory for output
 out:
 	@mkdir out || (exit 0)
 	mkdir out$/tests || (exit 0)
 
-
-.PHONY: clean build format imports lint test unit acceptance prepare-for-pr verify verify-format benchmark
```

Comment:
Why remove `.PHONY` entirely? Can it work with a comment in front, as you have above?

Reply:
Sorry, it's not included, because the first commit didn't pass, so I'll try to remove it in the second commit, I'll do it in the next commit